pkg/threescale/proxy_conf.go

```diff
@@ -274,7 +273,7 @@ func (pc *ProxyConfigCache) refreshCache() bool {
 	// check if any remote hosts have not been reachable
 	shouldRetry := !isEmptySet(pc.misbehavingHosts)
 	// reset the set of unreachable hosts
-	emptySet(pc.misbehavingHosts)
+	pc.misbehavingHosts = nil
```

Comment:
so I guess the issue was that we were setting to nil a _copy_ of the data structure?